server/src/lib/db.ts

```diff
@@ -158,7 +161,7 @@ export const saveUser = (
     userData: any,
 ): Promise<{_id: string}> => new Promise((resolve: Function, reject: Rejector) => {
     const users = db.collection('users');
-    users.insertOne(userData)
+    users.replaceOne({ _id: userData._id }, userData, { upsert: true })
```

Comment:
Might want to check whether `userData._id` exists? Otherwise `undefined` will be used as key